internal/fxclock/clock_test.go

```diff
@@ -22,44 +22,150 @@ package fxclock
 
 import (
 	"context"
+	"sync"
 	"testing"
 	"time"
 
-	"github.com/benbjohnson/clock"
 	"github.com/stretchr/testify/assert"
 )
 
-var _ Clock = clock.Clock(nil)
-
-// Just a basic sanity check that everything is in order.
```

Comment:
Instead of just the system clock sanity test, the test is now used for both, system clock and mock clock to verify basic outer expectations of the clocks.